cpp_linter/loggers.py

```diff
@@ -53,3 +54,32 @@ def log_response_msg(response: Response):
             response.request.url,
             response.text,
         )
+
+
+def worker_logfile_init(tempdir: str, loglvl: int):
+    logfile = NamedTemporaryFile("w", dir=tempdir, delete=False)
+
+    logger.handlers.clear()
+    logger.propagate = False
+
+    handler: logging.Handler
+    if FOUND_RICH_LIB:
+        console = get_console()
+        console.file = logfile
+        handler = RichHandler(show_time=False, console=console)
+        handler.setFormatter(logging.Formatter("%(name)s: %(message)s"))
+    else:
+        handler = logging.StreamHandler(logfile)
+        handler.setFormatter(logging.Formatter(logging.BASIC_FORMAT))
+    logger.addHandler(handler)
+    # Windows does not copy log level to subprocess.
+    # https://github.com/cpp-linter/cpp-linter/actions/runs/8355193931
+    logger.setLevel(loglvl)
+
+    log_commander.handlers.clear()
+    log_commander.propagate = False
+    console_handler = logging.StreamHandler(logfile)
+    console_handler.setFormatter(logging.Formatter("%(message)s"))
+    log_commander.addHandler(console_handler)
```

Comment:
With the log groups done in the main thread, I don't think we need to re-configure `log_commander`, right?

The clang-tidy.py and clan-format.py modules don't actually use `log_commander` (directly or otherwise).

Reply:
Correct, I just wanted to make sure everything to be captured. Maybe at some point log_commander could be used while running tools?

Reply:
> Maybe at some point log_commander could be used while running tools?

Its not likely. The log commander is only used to output github action log commands:
1. [grouping log lines](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines)
2. [making file annotations](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-notice-message)

I don't see any more usefulness [from those options](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions), especially during in threads when running clang-format and clang-tidy.

If you just want to comment the paragraph out, I'm ok with that too. Maybe github will add something that might be useful in the future. I just want to reduce the lines executed for each file's analysis.
```suggestion
    ## uncomment the following if log_commander is needed in isolated threads
    # log_commander.handlers.clear()
    # log_commander.propagate = False
    # console_handler = logging.StreamHandler(logfile)
    # console_handler.setFormatter(logging.Formatter("%(message)s"))
    # log_commander.addHandler(console_handler)
```